explore/src/main/scala/explore/constraints/ConstraintsPanel.scala

```diff
@@ -300,7 +304,12 @@ object ConstraintsPanel {
     }
 
     def render(props: Props, state: State) = AppCtx.using { implicit appCtx =>
-      renderFn(props, state, UndoContext(props.undoStacks, props.constraintSet))
+      val csUndoView: View[UndoStacks[IO, ConstraintSet]] =
+        props.undoStacks.zoom(_.groupStacks(props.obsIds))(mod =>
+          csus => ConstraintGroupUndoStacks(props.obsIds, mod(csus.groupStacks(props.obsIds)))
+        )
```

Comment:
Can we encapsulate the mod operations in `ConstraintsUndoStacks`? Maybe they pass enough laws to build an `Adjuster`? (And then we can zoom using a `GetAdjust`)

Reply:
I started doing that originally, but didn't know the right optic and was unsure about the legality - esp. with the group stack one requiring the observation ids. So, I decided to do it "in place" since each was used only once. But, I can check out an `Adjuster` and see if an implementation would be lawful.